contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,486 @@
+
+        // Unauthorized err
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::UpdateConfig { owner: None };
+
+        let res = execute(deps.as_mut(), env, info, msg);
+        match res {
+            Err(ContractError::Unauthorized { .. }) => {}
+            _ => panic!("Must return unauthorized error"),
+        }
+    }
+
+    #[test]
+    fn register_merkle_root() {
+        let mut deps = mock_dependencies(&[]);
+
+        let msg = InstantiateMsg {
+            owner: "owner0000".to_string(),
+            cw20_token_address: "anchor0000".to_string(),
+        };
+
+        let env = mock_env();
+        let info = mock_info("addr0000", &[]);
+        let _res = instantiate(deps.as_mut(), env, info, msg).unwrap();
+
+        // register new merkle root
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::RegisterMerkleRoot {
+            merkle_root: "634de21cde1044f41d90373733b0f0fb1c1c71f9652b905cdf159e73c4cf0d37"
+                .to_string(),
+        };
+
+        let res = execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+        assert_eq!(
+            res.attributes,
+            vec![
+                attr("action", "register_merkle_root"),
+                attr("stage", "1"),
+                attr(
+                    "merkle_root",
+                    "634de21cde1044f41d90373733b0f0fb1c1c71f9652b905cdf159e73c4cf0d37"
+                )
+            ]
+        );
+
+        let res = query(deps.as_ref(), env.clone(), QueryMsg::LatestStage {}).unwrap();
+        let latest_stage: LatestStageResponse = from_binary(&res).unwrap();
+        assert_eq!(1u8, latest_stage.latest_stage);
+
+        let res = query(
+            deps.as_ref(),
+            env,
+            QueryMsg::MerkleRoot {
+                stage: latest_stage.latest_stage,
+            },
+        )
+        .unwrap();
+        let merkle_root: MerkleRootResponse = from_binary(&res).unwrap();
+        assert_eq!(
+            "634de21cde1044f41d90373733b0f0fb1c1c71f9652b905cdf159e73c4cf0d37".to_string(),
+            merkle_root.merkle_root
+        );
+    }
+
+    #[test]
+    fn claim() {
+        let mut deps = mock_dependencies(&[]);
+
+        let msg = InstantiateMsg {
+            owner: "owner0000".to_string(),
+            cw20_token_address: "token0000".to_string(),
+        };
+
+        let env = mock_env();
+        let info = mock_info("addr0000", &[]);
+        let _res = instantiate(deps.as_mut(), env, info, msg).unwrap();
+
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::RegisterMerkleRoot {
+            merkle_root: "5d4f48f147cb6cb742b376dce5626b2a036f69faec10cd73631c791780e150fc"
+                .to_string(),
+        };
+        let _res = execute(deps.as_mut(), env, info, msg).unwrap();
+
+        let msg = ExecuteMsg::Claim {
```

Comment:
It would be good to document where these come from. Or even better, have a script that generates many of these examples, so we can use testvectors.

You can store as json files and include in the tests with `include_bytes!()`